atom/browser/atom_download_manager_delegate.h

```diff
@@ -39,25 +50,65 @@ class AtomDownloadManagerDelegate : public content::DownloadManagerDelegate {
       content::DownloadItem* download,
       const content::DownloadOpenDelayedCallback& callback) override;
   void GetNextId(const content::DownloadIdCallback& callback) override;
+  bool ShouldCompleteDownload(content::DownloadItem* item,
+                              const base::Closure& complete_callback) override;
+
+ protected:
+  virtual safe_browsing::DownloadProtectionService*
+      GetDownloadProtectionService();
+  void CheckDownloadUrl(content::DownloadItem* download,
+                            const base::FilePath& suggested_virtual_path,
+                            const CheckDownloadUrlCallback& callback) override;
+  void NotifyExtensions(content::DownloadItem* download,
+                        const base::FilePath& suggested_virtual_path,
+                        const NotifyExtensionsCallback& callback) override {}
+  void ReserveVirtualPath(
+      content::DownloadItem* download,
+      const base::FilePath& virtual_path,
+      bool create_directory,
+      DownloadPathReservationTracker::FilenameConflictAction conflict_action,
+      const ReservedPathCallback& callback) override;
+
+  void RequestConfirmation(content::DownloadItem* download,
```

Comment:
you can't override these and ignore the callback. That's likely why you need all the patches on `download_target_determiner.cc`